tests/testthat/_snaps/tree.md

```diff
@@ -0,0 +1,256 @@
+      ├─2:{list}
+      │ ├─id:"b"
+      │ ├─val:1
+      │ └─children:[list]
+      │   ├─1:{list}
+      │   │ ├─id:"b1"
+      │   │ └─val:2.5
+      │   └─2:{list}
+      │     ├─id:"b2"
+      │     ├─val:8
+      │     └─children:[list]
+      │       └─1:{list}
+      │         ├─id:"b21"
+      │         └─val:4
+      └─3:{list}
+        ├─id:"c"
+        ├─val:8
+        └─children:[list]
+          ├─1:{list}
+          │ └─id:"c1"
+          └─2:{list}
+            ├─id:"c2"
+            └─val:1
+    Code
+      tree(nested_lists, index_unnamed = FALSE)
+    Output
+      [list]
+      ├─{list}
+      │ ├─id:"a"
+      │ └─val:2
+      ├─{list}
+      │ ├─id:"b"
+      │ ├─val:1
+      │ └─children:[list]
+      │   ├─{list}
+      │   │ ├─id:"b1"
+      │   │ └─val:2.5
+      │   └─{list}
+      │     ├─id:"b2"
+      │     ├─val:8
+      │     └─children:[list]
+      │       └─{list}
+      │         ├─id:"b21"
+      │         └─val:4
+      └─{list}
+        ├─id:"c"
+        ├─val:8
+        └─children:[list]
+          ├─{list}
+          │ └─id:"c1"
+          └─{list}
+            ├─id:"c2"
+            └─val:1
+
+# Atomic arrays have sensible defaults with truncation added for longer than 10-elements
+
+    Code
+      tree(list(name = "vectored list", num_vec = 1:10, char_vec = letters))
+    Output
+      {list}
+      ├─name:"vectored list"
+      ├─num_vec:1,2,3,4,5,6,7,8,9,10
+      └─char_vec:"a","b","c","d","e","f","g","h","i","j",...(n = 26)
+
+# Works with HTML tag structures
+
+    Code
+      tree(shiny::sliderInput("test", "Input Label", 0, 1, 0.5))
+    Output
+      {shiny.tag}
+      ├─name:"div"
+      ├─attribs:{list}
+      │ └─class:"form-group shiny-input-container"
+      └─children:[list]
+        ├─1:{shiny.tag}
+        │ ├─name:"label"
+        │ ├─attribs:{list}
+        │ │ ├─class:"control-label"
+        │ │ ├─id:"test-label"
+        │ │ └─for:"test"
+        │ └─children:[list]
+        │   └─1:"Input Label"
+        └─2:{shiny.tag}
+          ├─name:"input"
+          ├─attribs:{list}
+          │ ├─class:"js-range-slider"
+          │ ├─id:"test"
+          │ ├─data-skin:"shiny"
+          │ ├─data-min:"0"
+          │ ├─data-max:"1"
+          │ ├─data-from:"0.5"
+          │ ├─data-step:"0.01"
+          │ ├─data-grid:"true"
+          │ ├─data-grid-num:10
+          │ ├─data-grid-snap:"false"
+          │ ├─data-prettify-separator:","
+          │ ├─data-prettify-enabled:"true"
+          │ ├─data-keyboard:"true"
+          │ └─data-data-type:"number"
+          └─children:[list]
+
+# Max depth and length can be enforced
+
+    Code
+      deep_list <- list(list(id = "b", val = 1, children = list(list(id = "b1", val = 2.5),
+      list(id = "b2", val = 8, children = list(list(id = "b21", val = 4))))), list(
+        id = "a", val = 2))
+      tree(deep_list, max_depth = 1)
+    Output
+      [list]
+      ├─1:{list}
+      │ ├─id:"b"
+      │ ├─val:1
+      │ └─children:[list]
```

Comment:
Maybe add `...` to more clearly indicate that there's more to see? (Same for `max_length`)